pkg/agent/run.go

```diff
@@ -161,11 +163,45 @@ func Run(cmd *cobra.Command, args []string) (returnErr error) {
 	if err != nil {
 		return fmt.Errorf("failed to create event recorder: %v", err)
 	}
+	// Data gatherers are loaded depending on what the Kubernetes API supports.
+	// First, let's do a /api discovery to see what the API supports.
+	discoveryClient, err := k8s.NewDiscoveryClient("")
+	if err != nil {
+		return fmt.Errorf("failed to create a discovery client: %v", err)
+	}
 
+	_, apiResourceLists, err := discoveryClient.ServerGroupsAndResources()
+	if err != nil {
+		return fmt.Errorf("failed to get server resources: %v", err)
+	}
+	availableGVRs := sets.NewString()
+	for _, arl := range apiResourceLists {
+		gv, err := schema.ParseGroupVersion(arl.GroupVersion)
+		if err != nil {
+			log.Error(err, "Failed to parse group version")
+			continue
+		}
+		for _, ar := range arl.APIResources {
+			availableGVRs.Insert(gv.WithResource(ar.Name).String())
+		}
+	}
+	log.V(logs.Debug).Info("API discovery complete", "found", availableGVRs.List())
 	dataGatherers := map[string]datagatherer.DataGatherer{}
 
 	// load datagatherer config and boot each one
 	for _, dgConfig := range config.DataGatherers {
+		if c, ok := dgConfig.Config.(*k8s.ConfigDynamic); ok {
+			gvr := c.GroupVersionResource
+			if !availableGVRs.Has(gvr.String()) {
```

Comment:
I think the challenge is that we want to start the DataGatherer when the CRD is installed at a later point in time.